hil/ext/switches/_console.py

```diff
@@ -178,3 +178,25 @@ def get_prompts(console):
             'if_prompt': re.escape(cmd_prompt[:-1]) + '\(config\\-if[^)]*\)#',
             'main_prompt': re.escape(cmd_prompt),
         }
+
+
+def login(switch):
+    """login to switch using either a password or key."""
```

Comment:
Should mention the requirement for username and password attributes on the switch; All of the switches we're using with this happen to have that attribute, but there's nothing defined by the Switch class that says they need to.